### DISCUSSION TITLE
Is my syntax for ignoring closed issues correct?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#27169] [Actions]
<p>I use the <code>peaceiris/actions-label-commenter</code> Action for giving canned responses to issues based on the labels I assign/remove.<br>
Now was my plan to add an <code>if: github.event.state == 'open'</code> to the job to only trigger it when the issue isn’t closed already to prevent unnesesary messages when someone labels the issue later on.</p>
<p>Issue (no pun intendet) is, that the Action seems to ignore pretty much any case, being it open or closed issues, which obviously is not what I want.</p>
<p>Here is the complete workflow:</p>
<pre><code class="lang-yaml">name: Comment
on:
  issues:
    types:
    - labeled
    - unlabeled
  pull_request:
    types:
    - labeled
    - unlabeled

jobs:
  give_comment:
    if: github.event.state == 'open'
    runs-on: ubuntu-latest
    steps:
    - name: Checkout Code
      uses: actions/checkout@v2
      with:
        ref: master
    - name: Send Issue/Pull request comment
      uses: peaceiris/actions-label-commenter@v1.8.2
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>According to the <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/issues" rel="noopener nofollow ugc">reference about issues</a> does the JSON object contain the “state” key which contains either <code>open</code>, <code>closed</code> or <code>all</code> (What even does the last one indicate for a state?)</p>
<p>Should I update my Workflow to have a negated if for when issues are closed (<code>if: github.event.state != 'closed'</code>) or is there something else I could do?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>I believe it’s <code>github.event.issue.state</code> that you need.</p>
<p>Actions uses webhook payloads when triggered and the <code>state</code> is listed inside the <code>issue</code> object of those:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#issues" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#issues" target="_blank" rel="noopener">Webhook events and payloads - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#issues" target="_blank" rel="noopener">//docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by Andre601** [comment]
<p>Didn’t saw it being inside a issues object, so it makes sense now that it didn’t work.</p>
